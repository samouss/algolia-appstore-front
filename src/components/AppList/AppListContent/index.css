.AppListContent {
  flex: 1;
  border: 1px solid #f2f2f2;
  background-color: #ffffff;
}
