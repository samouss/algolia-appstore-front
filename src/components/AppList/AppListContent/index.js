import React from 'react';
import PropTypes from 'prop-types';
import InfiniteList from 'react-infinite-scroll-list';
import AppItem, { HitPropTypes } from 'components/AppItem';
import './index.css';

const AppListContent = ({
  hits,
  isLoading,
  isEndReached,
  onNextPage,
}) => (
  <InfiniteList
    styleName="AppListContent"
    containerTagName="ul"
    isLoading={isLoading}
    isEndReached={isEndReached}
    isAttachOnWindow
    onReachThreshold={onNextPage}
    threshold={680}
  >
    {hits.map(hit => (
      <AppItem
        key={hit.objectID}
        name={hit.name}
        image={hit.image}
        link={hit.link}
        rating={hit.rating}
        ratingCount={hit.ratingCount}
        price={hit.price}
        _highlightResult={hit._highlightResult}
      />
    ))}
  </InfiniteList>
);

AppListContent.propTypes = {
  hits: PropTypes.arrayOf(PropTypes.shape(HitPropTypes)).isRequired,
  isLoading: PropTypes.bool.isRequired,
  isEndReached: PropTypes.bool.isRequired,
  onNextPage: PropTypes.func.isRequired,
};

export default AppListContent;
