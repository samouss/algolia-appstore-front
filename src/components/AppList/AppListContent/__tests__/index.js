import React from 'react';
import { shallow } from 'enzyme';
import AppListContent from '../index';

describe('<AppListContent />', () => {
  const defaultProps = {
    hits: [
      {
        objectID: '1234',
        name: 'Cut the Rope',
        image: 'http://image.com',
        link: 'http://link.com',
        rating: 3,
        ratingCount: 1234,
        price: '0 $',
        _highlightResult: {
          name: {
            value: '__algolia__highlight__pre__tag__Cut the__algolia__highlight__post__tag__ Rope',
          },
        },
      },
      {
        objectID: '5678',
        name: 'Angry Birds',
        image: 'http://image.com',
        link: 'http://link.com',
        rating: 5,
        ratingCount: 6789,
        price: '0 $',
        _highlightResult: {
          name: {
            value: 'Angry Birds',
          },
        },
      },
    ],
    isLoading: false,
    isEndReached: false,
    onNextPage: () => {},
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <AppListContent
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onNextPage', () => {
    const props = {
      ...defaultProps,
      onNextPage: jest.fn(),
    };

    const component = shallow(
      <AppListContent
        {...props}
      />,
    );

    component
      .find('InfiniteList')
      .simulate('reachThreshold');

    expect(props.onNextPage).toHaveBeenCalled();
  });
});
