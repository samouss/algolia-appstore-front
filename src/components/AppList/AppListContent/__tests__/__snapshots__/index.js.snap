// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppListContent /> expect to render 1`] = `
<InfiniteList
  className="infinite-list"
  containerHeight=""
  containerTagName="ul"
  isAttachOnWindow={true}
  isEndReached={false}
  isLoading={false}
  onReachThreshold={[Function]}
  styleName="AppListContent"
  threshold={1700}
>
  <AppItem
    _highlightResult={
      Object {
        "name": Object {
          "value": "__algolia__highlight__pre__tag__Cut the__algolia__highlight__post__tag__ Rope",
        },
      }
    }
    image="http://image.com"
    link="http://link.com"
    name="Cut the Rope"
    price="0 $"
    rating={3}
    ratingCount={1234}
  />
  <AppItem
    _highlightResult={
      Object {
        "name": Object {
          "value": "Angry Birds",
        },
      }
    }
    image="http://image.com"
    link="http://link.com"
    name="Angry Birds"
    price="0 $"
    rating={5}
    ratingCount={6789}
  />
</InfiniteList>
`;
